doc/book/admin/access_control.rst

```diff
@@ -1,270 +1,865 @@
 .. _authentication:
+.. _access_control:
 
-================================================================================
 Access control
-================================================================================
+==============
 
-This section explains how Tarantool makes it possible for administrators
-to prevent unauthorized access to the database and to certain functions.
+Tarantool enables flexible management of access to various database resources.
+The main concepts of Tarantool access control system are as follows:
 
-Briefly:
+*   A *user* is a person or program that interacts with a Tarantool instance.
+*   An *object* is an entity to which access can be granted, for example, spaces, indexes, or functions.
+*   *Privileges* allow a user to perform certain operations on specific objects, for example, creating spaces, reading or updating data.
+*   A *role* is a named collection of privileges that can be granted to a user.
 
-* There is a method to guarantee with password checks that users really are
-  who they say they are (“authentication”).
 
-* There is a :ref:`_user <box_space-user>` system space, where usernames and
-  password-hashes are stored.
 
-* There are functions for saying that certain users are allowed to do certain
-  things (“privileges”).
+.. _access_control_overview:
 
-* There is a :ref:`_priv <box_space-priv>` system space, where privileges are
-  stored. Whenever a user tries to do an operation, there is a check whether
-  the user has the privilege to do the operation (“access control”).
-
-Details follow.
+Overview
+--------
 
 .. _authentication-users:
+.. _access_control_concepts_users:
 
---------------------------------------------------------------------------------
 Users
---------------------------------------------------------------------------------
+~~~~~
 
-There is a **current user** for any program working with Tarantool,
-local or remote.
-If a remote connection is using a :ref:`binary port <admin-security>`,
-the current user, by default, is '**guest**'.
-If the connection is using an :ref:`admin-console port <admin-security>`,
-the current user is '**admin**'.
-When executing a :ref:`Lua initialization script <index-init_label>`,
-the current user is also ‘**admin**’.
+A user is a person or program that interacts with a Tarantool instance.
+There might be different types of users, for example:
 
-The current user name can be found with
-:doc:`/reference/reference_lua/box_session/user`.
+*   A database administrator is responsible for the overall management and administration of a database.
+    An administrator can create other users and grant them specified privileges.
+*   A regular user has limited access to certain data and stored functions. For example, such users can be used to let :ref:`connectors <index-box_connectors>` communicate to a database.
```

Comment:
Good point. Removed information about connectors because this is a specific case. Both admin and regular user might work directly or using a connector.
Also, added info that users can get privileges from dbadmin.